client/signup/wpcom-login-form/index.jsx

```diff
@@ -4,6 +4,22 @@ import { Component } from 'react';
 
 const debug = debugFactory( 'calypso:signup:wpcom-login' );
 
+function getFormAction( redirectTo ) {
+	const subdomainRegExp = /^https?:\/\/([a-z0-9]*).wordpress.com/;
+	const hostname = config( 'hostname' );
+	let subdomain = '';
+
+	if (
+		subdomainRegExp.test( redirectTo ) &&
+		hostname !== 'wpcalypso.wordpress.com' &&
+		hostname !== 'horizon.wordpress.com'
+	) {
+		subdomain = redirectTo.match( subdomainRegExp )[ 1 ] + '.';
+	}
+
+	return `https://${ subdomain }wordpress.com/wp-login.php`;
+}
+
 export default class WpcomLoginForm extends Component {
```

Comment:
After inlining the `renderExtraFields` method and converting `componentDidMount` to `useEffect` we could convert this class component to functional.